### PR TITLE
[modbus] Validate poller count and write count according to modbus spec

### PR DESCRIPTION
**Pull request by ssalonen**
Related to discussion in #8228

Signed-off-by: Sami Salonen <ssalonen@gmail.com>

### CODE REVIEW

**Comment by ssalonen** [comment]
WIP. Needs to be tested with real setup as well. Especially data thing going offline/online properly when poller is invalid/valid.

File: bundles/org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/handler/ModbusPollerThingHandler.java [hunk 1]
```diff
@@ -204,6 +178,12 @@ public String toString() {
 
     private final Logger logger = LoggerFactory.getLogger(ModbusPollerThingHandler.class);
 
+    private final static Object[] SORTED_READ_FUNCTION_CODES = ModbusBindingConstantsInternal.READ_FUNCTION_CODES
+            .keySet().toArray();
+    static {
+        Arrays.sort(SORTED_READ_FUNCTION_CODES);
+    }
```
Comment (Hilbrand) [inline bundles/org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/handler/ModbusPollerThingHandler.java:185]:
```suggestion
    private final static List<String> SORTED_READ_FUNCTION_CODES = ModbusBindingConstantsInternal.READ_FUNCTION_CODES.keySet().stream().sorted().collect(Collectors.toList());
```

**Comment by ssalonen** [comment]
Travis failed due spotless issues, need to run the formatting.

**Review by Hilbrand — APPROVED** [review]
LGTM

**Comment by Hilbrand** [comment]
When you're done testing, please remove the work-in-progress and we can merge this pr.

**Comment by ssalonen** [comment]
According to further testing and reading the spec. Limit is in fact 123 (not 125 like in the PR) registers with modbus tcp. Spec says with RTU it is 125. We might simply limit all to 123. Need to change and document this.

Online/offline thing handling seems to work well itself. Poller status is propagated correctly to data thing.

**Comment by ssalonen** [comment]
This is good to merge once the tests pass.

**Pull request merged**